src/core/exec-invoke.c

```diff
@@ -2280,6 +2280,56 @@ static int create_many_symlinks(const char *root, const char *source, char **sym
         return 0;
 }
 
+static int is_idmapping_supported(const char *path) {
+        _cleanup_close_ int mount_fd = -EBADF, userns_fd = -EBADF, dir_fd = -EBADF;
+        _cleanup_free_ char *uid_map = NULL, *gid_map = NULL;
+        int r;
+
+        assert(path);
+
+        dir_fd = open(path, O_RDONLY|O_CLOEXEC|O_DIRECTORY|O_NOFOLLOW);
+        if (dir_fd < 0) {
+                if (ERRNO_IS_NOT_SUPPORTED(errno) || errno == EINVAL || errno == ENOTDIR)
+                        return false;
+                return -errno;
+        }
+
+        mount_fd = open_tree(dir_fd, "", AT_EMPTY_PATH | OPEN_TREE_CLONE | OPEN_TREE_CLOEXEC);
+        if (mount_fd < 0) {
+                if (ERRNO_IS_NOT_SUPPORTED(errno) || errno == EINVAL)
+                        return false;
+                return -errno;
+        }
+
+        r = strextendf(&uid_map, UID_FMT " " UID_FMT " " UID_FMT "\n", UID_NOBODY, UID_NOBODY, 1u);
+        if (r < 0)
+                return r;
+
+        r = strextendf(&gid_map, GID_FMT " " GID_FMT " " GID_FMT "\n", GID_NOBODY, GID_NOBODY, 1u);
+        if (r < 0)
+                return r;
+
+        userns_fd = userns_acquire(uid_map, gid_map);
+        if (userns_fd < 0) {
+                if (ERRNO_IS_NEG_NOT_SUPPORTED(userns_fd))
+                        return false;
+                return userns_fd;
+        }
+
+        r = mount_setattr(mount_fd, "", AT_EMPTY_PATH,
+                          &(struct mount_attr) {
+                                .attr_set = MOUNT_ATTR_IDMAP | MOUNT_ATTR_NOSUID | MOUNT_ATTR_NOEXEC,
```

Comment:
doesn't really matter, but maybe also add MOUNT_ATTR_RDONLY here, and NODEV? i mean, this is shortlived and doesn#t really matter, but let's just lock things down as much as we can as a matter of coding style, always